django/template/defaultfilters.py

```diff
@@ -126,13 +126,28 @@ def floatformat(text, arg=-1):
     * {{ 6666.6666|floatformat:"2g" }} displays "6,666.67"
     * {{ 10000|floatformat:"g" }} displays "10,000"
 
+    If arg has the 'u' suffix, force the result to be unlocalized. When the
+    active locale is pl (Polish):
+
+    * {{ 66666.6666|floatformat:"2" }} displays "66666,67"
+    * {{ 66666.6666|floatformat:"2u" }} displays "66666.67"
+
     If the input float is infinity or NaN, display the string representation
     of that value.
     """
     force_grouping = False
-    if isinstance(arg, str) and arg.endswith('g'):
-        force_grouping = True
-        arg = arg[:-1] or -1
+    use_l10n = True
+    if isinstance(arg, str):
+        if arg.endswith(('gu', 'ug')):
+            force_grouping = True
+            use_l10n = False
+            arg = arg[:-2] or -1
+        elif arg.endswith('g'):
+            force_grouping = True
+            arg = arg[:-1] or -1
+        elif arg.endswith('u'):
+            use_l10n = False
+            arg = arg[:-1] or -1
```

Comment:
Just throwing it out there; for floatformat uses which don't include `g` or `u` they're paying the cost for exhaustive checking. We can minimise that cost if you're bothered about it.

```
    last_chars = arg[-2:]
    last_char = arg[-1]
    if last_chars in {'ug', 'gu'}:
        force_grouping = True
        use_l10n = False
        arg = arg[:-2] or -1
    elif last_char == 'g':
        force_grouping = True
        arg = arg[:-1] or -1
    elif last_char == 'u':
        use_l10n = False
        arg = arg[:-1] or -1
```

eg: Back of the napkin costs; if I wrap the original `if` branches into a function `endswiths` (which does nothing else), and the ones above into `charslice`:

```
# Unsuffixed, most common usage presumably.
In [2]: %timeit endswiths('21525.3532')
450 ns ± 5.76 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
In [3]: %timeit charslice('21525.3532')
305 ns ± 7.34 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

# Both suffixes
In [4]: %timeit endswiths('21525.3532gu')
322 ns ± 4.62 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
In [5]: %timeit charslice('21525.3532gu')
350 ns ± 10.2 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

# u suffix
In [6]: %timeit endswiths('21525.3532u')
529 ns ± 8.17 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
In [7]: %timeit charslice('21525.3532u')
385 ns ± 6.24 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

# g suffix
In [8]: %timeit endswiths('21525.3532g')
435 ns ± 6.33 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
In [9]: %timeit charslice('21525.3532g')
355 ns ± 3.67 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```